add a query to retrieve taxon IDs for entities.

sinistral opened the pull request:
Example:

```
>> java -jar target/kabob-query-0.1.0-SNAPSHOT-standalone.jar kabob-query/cli -q sparql/taxonomy/ncbi-id -a "[\"uniprot:J3KMH5\"]" -p "{:db-url \"http://amc-tantor.ucdenver.pvt:10035\" :repository-name \"kabob-dev\" :username \"user\" :password \"****\"}"
entity_id, ncbi_id
uniprot:J3KMH5, obo:NCBITaxon_10090
```

sinistral committed:
Add a query to retrieve taxon IDs for entities.

bill-baumgartner commented on resources/sparql/taxonomy/ncbi-id.mustache:
```diff
@@ -0,0 +1,12 @@
+# Given a bio entity, return the NCBI taxonomy ID to which it belongs.
+select ("{{src-id}}" as ?entity_id) ?ncbi_id
+where {
+  <{{bio-id}}> obo:RO_0002162 ?ncbi_id
```
To be safe, I think we should be referencing the restriction that is onProperty RO_0002162 and someValuesFrom ?ncbi_id.  As written, the query works but relies on what appears to be a shortcut relation that we should probably remove from KaBOB.

sinistral replied:
Good point, thanks.  I didn't know about the restriction.  
Learning question: how did you identify that as a shortcut relationship?

sinistral committed:
Identify the taxonomy by a restriction.

The direct reference `in taxon` [1] link between the BIO entity and the
taxon type is believed to be a 'shortcut' relation that should
eventually be removed from the store.  Thus we use the property
restriction to identify the taxonomy ID.

[1] http://purl.obolibrary.org/obo/RO_0002162

sinistral commented:
I've updated this pull request with a change to determine the taxonomy ID using the restriction.
